File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        map<int, int> checkSum;
        for(int i = 0; i < nums.size() - 1; ++i){
            int diff = target - nums[i];
            checkSum[diff] = i;
        }
        for(int j = 0; j < nums.size(); ++j){
            int prevNum = nums[checkSum[nums[j]]];
            if (checkSum.count(nums[j]) && prevNum + nums[j] == target && j != checkSum[nums[j]]){
                result.push_back(checkSum[nums[j]]);
                result.push_back(j);
                return result;
            }
        }
        return result;
    }

};

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let result = [];
    let checkSum = {};
    for(let i = 0; i < nums.length; i++){
        let diff = target - nums[i];
        checkSum[diff] = i;
    }
    for(let j = 0; j < nums.length; j++){
        let prevNum = nums[checkSum[nums[j]]];
        if (prevNum + nums[j] == target && j != checkSum[nums[j]]){
            result.push(checkSum[nums[j]]);
            result.push(j);
            return result;
        }
    }
    return result;
};